### PR TITLE
ENT-3218: Setting up SonarQube for Candlepin

### PR DESCRIPTION
**Pull request by sonalidhome**
- Updated the dependencies for SonarQube
 - Added a Jenkins job to generate a report on a sonarqube server

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -119,6 +119,37 @@ pipeline {
...
+                        checkout scm
+                        sh 'docker login -u "$CANDLEPIN_QUAY_BOT_USR" -p "$CANDLEPIN_QUAY_BOT_PSW" quay.io'
+                        sh 'sh jenkins/upload-on-sonarqube.sh'
+                        sh 'docker logout quay.io'
+                    }
+                }
+                stage('Upload-Branch-to-SonarQube') {
+                    agent { label 'docker' }
+                    environment {
+                        SONAR_HOST_URL = 'https://sonarqube.corp.redhat.com'
```
Comment (nikosmoum) [inline jenkins/Jenkinsfile:138]: Please remove the url from the code and the commit message, and use a Jenkins secret to pass this in

Comment (sonalidhome) [reply jenkins/Jenkinsfile]: Ack. I missed this configuration.

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -119,6 +119,37 @@ pipeline {
                         sh 'docker logout quay.io'
                     }
                 }
+                stage('Upload-PR-to-SonarQube') {
+                    agent { label 'docker' }
+                    environment {
+                        SONAR_HOST_URL = credentials('CANDLEPIN_SONAR_HOST_URL')
```
Comment (Lorquas) [inline jenkins/Jenkinsfile:125]: I added this `SONAR_HOST_URL` as a global env var, you can use it directly and we don't have to do this credentials lookup.

File: docker/docker-compose-postgres.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ services:
     privileged: true
     volumes:
       - ../:/candlepin-dev
+      - ${SONAR_CRT:-../server/conf/candlepin-redhat-ca.crt}:/candlepin-dev/certs/SONAR.crt
```
Comment (sonalidhome) [inline docker/docker-compose-postgres.yml:26]: @Lorquas, Updated the code to access the `SONAR_CRT` as a global environment variable for the cert path. 
Can you add the cert path into the Jenkins env?

**Comment by nikosmoum** [comment]
Closing this since it is becoming stale. Once the prerequisite work is done and we can work on this, we can re-open or open a new one.

**Pull request closed without merging**